### PR TITLE
feat(stepfunctions-tasks): AWS SDK service integrations

### LINKED ISSUE #16780: (aws-stepfunctions-tasks): support AWS SDK service integrations
### Description

StepFunctions now integrates with over 200 AWS Services via [AWS SDK service integrations](https://docs.aws.amazon.com/step-functions/latest/dg/supported-services-awssdk.html). It would be great to have a higher level construct than the `CustomState` to use these

### Use Case

Build step functions using the new service integrations and thereby removing a lot of lambdas

### Proposed Solution

Something like:

```ts
new tasks.AwsSdk(this, 'Call S3', {
  service: "s3",
  action: "putObject",
  payload: sfn.TaskInput.fromObject({
    Bucket: sfn.JsonPath.stringAt("$.bucketName"),
    Body: "Hello World",
    Key: "test.txt"
  })
});
```

### Other information

_No response_

### Acknowledge

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by jogold**
Add support for Step Functions' AWS SDK integrations to call
any of the over two hundred AWS services directly from a state
machine.

See https://docs.aws.amazon.com/step-functions/latest/dg/supported-services-awssdk.html
See https://aws.amazon.com/blogs/aws/now-aws-step-functions-supports-200-aws-services-to-enable-easier-workflow-automation/

Closes #16780 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
@kaizen3031593

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  constructor(scope: Construct, id: string, private readonly props: CallAwsServiceProps) {
+    super(scope, id, props);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: props.iamResources,
+        actions: [props.iamAction ?? `${props.service}:${props.action}`],
```
Comment (RomainMuller) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts:53]: Is the default not guaranteed to fail here? The `action` is in the SFN integration is passed as `camelCase`, but IAM expects `PascalCase`...

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn(
+        'aws-sdk',
+        `${this.props.service.toLowerCase()}:${this.props.action}`,
```
Comment (RomainMuller) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts:65]: This is bound to have a surprising behavior (read: broken). if `this.props.service` is an unresolved token.

Comment (jogold) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/aws-sdk/call-aws-service.ts:65]: Ok, will make this token proof.

**Pull request merged**